### ISSUE TITLE
SPARK-21513: to_json support map[struct, struct] to json

### ISSUE THREAD

**Issue by Feng-Jiang28** [NVIDIA/spark-rapids#10918]

**Comment by revans2** [comment]
I don't think we want to try and support having anything but a String as the key for a map. At least not until we have a customer use case that asks for it. I know in the past toString was called on the java class that was the key and there were inconsistent/no-sensical results at best.

**Comment by revans2** [comment]
After https://github.com/NVIDIA/spark-rapids/pull/11642 this no longer throws an exception. Instead it falls back to the CPU.
